app/javascript/components/Admin/Purchase.tsx

```diff
@@ -0,0 +1,737 @@
+          <AdminActionButton
+            label="Cancel Pre-order"
+            url={Routes.cancel_preorder_by_seller_path(purchase.external_id)}
+            loading="Canceling..."
+            done="Cancelled!"
+            confirm_message="Are you sure you want to cancel this preorder?"
+            success_message="Cancelled!"
+          />
+        ) : null}
+
+        {purchase.subscription ? (
+          <>
+            <dt>Cancelled</dt>
+            <dd>
+              {purchase.subscription.cancelled ? (
+                <>
+                  <YesIcon />
+                  (on {purchase.subscription.cancelled.at} by{" "}
+                  {purchase.subscription.cancelled.by_buyer ? "buyer" : "seller"})
+                </>
+              ) : (
+                <NoIcon />
+              )}
+            </dd>
+
+            <dt>Ended</dt>
+            <dd>
+              {purchase.subscription.ended_at ? (
+                <>
+                  <YesIcon />
+                  (on {purchase.subscription.ended_at})
+                </>
+              ) : (
+                <NoIcon />
+              )}
+            </dd>
+
+            <dt>Failed</dt>
+            <dd>
+              {purchase.subscription.failed_at ? (
+                <>
+                  <YesIcon />
+                  (on {purchase.subscription.failed_at})
+                </>
+              ) : (
+                <NoIcon />
+              )}
+            </dd>
+          </>
+        ) : null}
+
+        {purchase.license ? (
+          <>
+            <dt>License</dt>
+            <dd>{purchase.license.serial}</dd>
+          </>
+        ) : null}
+
+        {purchase.affiliate_email ? (
+          <>
+            <dt>Affiliate</dt>
+            <dd>
+              <Link href={Routes.admin_search_users_path({ query: purchase.affiliate_email })}>
+                {purchase.affiliate_email}
+              </Link>
+            </dd>
+          </>
+        ) : null}
+
+        {purchase.refund_policy ? (
+          <>
+            <dt>Refund Policy</dt>
+            <dd>
+              <strong>
+                <RefundPolicyTitle refundPolicy={purchase.refund_policy} />
+                {purchase.refund_policy.fine_print ? (
+                  <>
+                    <br />
+                    {purchase.refund_policy.fine_print}
+                  </>
+                ) : null}
+                {purchase.refund_policy.max_refund_period_in_days ? (
+                  <>
+                    <br />
+                    Max refund period: {purchase.refund_policy.max_refund_period_in_days} days
+                  </>
+                ) : null}
+              </strong>
+            </dd>
+          </>
+        ) : null}
+        <dt>Can email</dt>
+        <dd aria-label="Can email">
+          <BooleanIcon value={purchase.can_contact} />
+        </dd>
+      </dl>
+    </div>
+    {purchase.gift ? (
+      <>
+        <hr />
+        {purchase.gift.is_sender_purchase ? (
```

Comment:
The same info is listed twice in [the original](https://github.com/antiwork/gumroad/blob/c8da59289076c4774a6552e6cd01cff4489b6061/app/views/admin/purchases/_purchase.html.erb#L281), so I removed one. Lmk if you think I missed something.

I also based this code on the assumption that a purchase can't be _both_ a sender and receiver purchase, which should always be true based on the code I looked at. If you can think of an instance where this would be problematic, lmk.